### PR TITLE
authn: default peer authn to a PERMISSIVE is none effective exists

### PR DESCRIPTION
**Pull request by kailun-qin**
Previously, `composePeerAuthentication` returns nil which can be used to
indicate no applicable (beta) policy exist in order to trigger fallback to
alpha policy if the input config list is empty. This can be simplified as we've
deprecated alpha policy.

This patch defaults the peer authn to a PERMISSIVE if no effective config input.
It helps clarify the behavior when no beta peer authn policy applied.

Related: https://github.com/istio/istio/issues/22602

Signed-off-by: Kailun Qin <kailun.qin@intel.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by kailun-qin** [comment]
@yangminzhu PTAL, thanks!

File: tests/integration/security/testdata/beta-mtls-permissive.yaml [hunk 1]
```diff
@@ -1,15 +1,3 @@
-# Global PeerAuthentication can be removed for this test, once we remove the (alpha) mesh policy
```
Comment (yangminzhu) [inline tests/integration/security/testdata/beta-mtls-permissive.yaml:1]: Can you add a new test case instead of changing the existing one? Should be very simple to just duplicate the test you're changing here.

File: pilot/pkg/security/authn/v1beta1/policy_applier.go [hunk 2]
```diff
@@ -486,19 +492,6 @@ func composePeerAuthentication(rootNamespace string, configs []*config.Config) *
 		}
 	}
 
-	if meshCfg == nil && namespaceCfg == nil && workloadCfg == nil {
-		// Return nil so that caller can fallback to apply alpha policy. Once we deprecate alpha API,
-		// this special case can be removed.
-		return nil
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier.go:492]: Could you also update other places (e.g. `getMutualTLSModeForPort`) that are using the result from here so that they do not need to check nil now?

File: tests/integration/security/testdata/beta-mtls-permissive.yaml (hunk 1 above)
Comment (kailun-qin) [reply tests/integration/security/testdata/beta-mtls-permissive.yaml]: Thanks, updated!

File: pilot/pkg/security/authn/v1beta1/policy_applier.go (hunk 2 above)
Comment (kailun-qin) [reply pilot/pkg/security/authn/v1beta1/policy_applier.go:492]: Good catch, removed related nil checks, thanks!

File: pilot/pkg/security/authn/v1beta1/policy_applier_test.go [hunk 3]
```diff
@@ -1791,7 +1795,7 @@ func TestComposePeerAuthentication(t *testing.T) {
 			},
 		},
 		{
-			name: "ignore non-empty selector in root namespace",
+			name: "return non-empty selector in root namespace with default",
```
Comment (yangminzhu) [inline pilot/pkg/security/authn/v1beta1/policy_applier_test.go:1798]: hmm, the name is a bit confusing, I think you can keep the old name, it's still to ignore the non-empty selector but just return permissive instead of nil.

**Review by yangminzhu — COMMENTED** [review]
Looks good! Just one small comment, please also update to merge the latest.

File: pilot/pkg/security/authn/v1beta1/policy_applier_test.go (hunk 3 above)
Comment (kailun-qin) [reply pilot/pkg/security/authn/v1beta1/policy_applier_test.go]: Thanks! Changed back to the original naming and rebased to master.

**Review by yangminzhu — APPROVED** [review]
